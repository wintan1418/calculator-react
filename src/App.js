const App = () => {
  const buttonColor = "#f2a33c";
  return (
    <div className="App">
      <div className = "calc-wrapper">
        <Input/>
        <div className = "row">
          <Button symbol = "AC"/>
          <Button symbol = "+/-"/>
          <Button symbol = "%"/>
          <Button symbol = "รท" color={buttonColor}/>
        </div>
