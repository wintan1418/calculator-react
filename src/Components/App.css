body {
  background: rgba(82, 67, 8, 0.589);
}

.calculator {
  width: 700px;
  display: flex;
  flex-direction: column;
}

.navbar {
  background: rgb(87, 85, 3);
  border-radius: 5%;
}

.navbar h2 {
  margin-top: 25px;
  color: rgba(12, 94, 12, 0.671);
  margin-left: 20px;
}

.navbar ul li {
  display: inline;
  margin-left: 20px;
}

.navbar ul li:hover {
  color: green;
}

.title {
  margin-left: 30px;
  color: green;
}

.calculator .home {
  margin-left: 20px;
  font-size: 19px;
}

.quote {
  color: orange;
}

.day-quote {
  font-size: 31px;
}

.day-quote span {
  color: white;
}

.home-title {
  margin-top: -410px;
}

.text {
  margin-left: 20px;
  font-size: 20px;
}
