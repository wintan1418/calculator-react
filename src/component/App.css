body {
  background-size: cover;
}

.calculator {
  width: 700px;
  display: flex;
  flex-direction: column;
}
