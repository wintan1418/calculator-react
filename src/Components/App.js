import React from 'react';
import calculate from '../logic/calculate';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import './App.css';

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      total: null,
      next: null,
      operation: null,
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(name) {
    this.setState(state => calculate(state, name));
  }

  get result() {
    const { next, operation, total } = this.state;

    return `${next || ''}${operation || ''}${total || ''}`;
  }

  render() {
    return (
      <div className="calculator">
        <Display result={this.result} />
        <ButtonPanel clickHandler={this.handleClick} />
      </div>
    );
  }
}

export default App;
