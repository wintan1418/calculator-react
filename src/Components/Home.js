import React from 'react';
import './App.css';

const Home = () => (
  <>