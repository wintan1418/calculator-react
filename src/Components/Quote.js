import React from 'react';
import './App.css';

const Quote = () => (
  <>
    <div className="quote-nav" />
    <div className="calculator">
      <h1 className="quote">Quote of the day</h1>
      <p className="day-quote">
        <q>
          Without mathematics, Everything around you is numbers.
          <span>Albert Einstein</span>
        </q>
      </p>

    </div>
  </>
);

export default Quote;
