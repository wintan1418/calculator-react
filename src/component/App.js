import '../App.css';
import Display from './Display';
import ButtonShape from './ButtonShape';
import React from 'react';

class App extends React.Component{
 constructor()[
   
 ]
};


function App() {
  return (
    <>
      <Display />
      <ButtonShape />
    </>
  );
}

export default App;
