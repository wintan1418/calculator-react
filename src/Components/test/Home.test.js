import React from 'react';
import ReactDOM from 'react-dom';
import { create } from 'react-test-renderer';
import Quote from '../Quote';
import Home from '../Home';
import App from '../App';

describe('Home  Componenet', () => {
  it('render without craching', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Quote />, div);
  });

  it('render without craching', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Home />, div);
  });

  it('renders correctly', () => {
    const quote = create(<Quote />).toJSON();
    expect(quote).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const home = create(<Home />).toJSON();
    expect(home).toMatchSnapshot();
  });

  // to not get confused I add this comment this is the calculator it's inside the app
  it('renders correctly', () => {
    const calculator = create(<App />).toJSON();
    expect(calculator).toMatchSnapshot();
  });
});
