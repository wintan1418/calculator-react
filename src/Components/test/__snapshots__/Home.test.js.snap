// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home  Componenet renders correctly 1`] = `
Array [
  <div
    className="quote-nav"
  />,
  <div
    className="calculator"
  >
    <h1
      className="quote"
    >
      Quote of the day
    </h1>
    <p
      className="day-quote"
    >
      <q>
        Without mathematics, there’s nothing you can do. Everything around you is mathematics. Everything around you is numbers..
        <span>
          – Shakuntala Devi
        </span>
      </q>
    </p>
  </div>,
]
`;

exports[`Home  Componenet renders correctly 2`] = `
<div
  className="calculator"
>
  <h2
    className="title"
  >
    Welcome to Wintan calculator
  </h2>
  <div
    className="text"
  >
    There are 26 letters of the English alphabet, but there are 44 sounds of English. Kids learn letters, but they also need to learn phonemes and graphemes. And once they’ve gotten that, they need to segment, blend, and manipulate sounds. Let’s break all that down and look at some ways to teach the 44 sounds of English.Phonemes are basic units of sounds. There are 44 phonemes that can be blended to form additional sounds.Graphemes are the way we write phonemes. You can have a one letter grapheme or multi-letter grapheme. For example, digraphs are two-letter graphemes that represent one phoneme or two letters that represent one sound. You can also have three- or four-letter graphemes.
  </div>
</div>
`;

exports[`Home  Componenet renders correctly 3`] = `
<div
  className="calculator"
>
  <div
    className="display"
  >
    
  </div>
  <div
    className="button-panel"
  >
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#ffa500",
            "width": "25%",
          }
        }
        type="button"
      >
        ÷
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#ffa500",
            "width": "25%",
          }
        }
        type="button"
      >
        x
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#ffa500",
            "width": "25%",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#ffa500",
            "width": "25%",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "50%",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#E0E0E0",
            "width": "25%",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#ffa500",
            "width": "25%",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
